src/Sentry.DiagnosticSource/Internals/DiagnosticSource/SentrySqlListener.cs

```diff
@@ -90,7 +90,7 @@ private void AddSpan(SentrySqlSpanType type, string operation, KeyValuePair<stri
         {
             _hub.ConfigureScope(scope =>
             {
-                if (scope.Transaction is not { } transaction)
+                if (scope.Transaction is not {} transaction || transaction.IsSampled != true)
```

Comment:
I'm not certain, but should we actually be doing this check here? I though the sampling decision was made later.

Reply:
I did some research and I was mistaken.  The sampling decision happens as the transaction is *started*.  So checking it here is an optimization, to not go to the effort of adding spans when we already know the transaction won't be sent anyway.